crates/consensus/src/receipt/receipts.rs

```diff
@@ -186,6 +184,22 @@ impl<T> FromIterator<Vec<T>> for Receipts<T> {
     }
 }
 
+impl<T: Encodable> Encodable for Receipts<T> {
+    fn encode(&self, out: &mut dyn BufMut) {
+        self.receipt_vec.encode(out)
+    }
+
+    fn length(&self) -> usize {
+        self.receipt_vec.length()
+    }
+}
+
+impl<T: Decodable> Decodable for Receipts<T> {
+    fn decode(buf: &mut &[u8]) -> alloy_rlp::Result<Self> {
+        Ok(Self { receipt_vec: Decodable::decode(buf)? })
+    }
+}
```

Comment:
aren't those derivable?

Reply:
Yes this should mimic the rlpx format. I'm not sure how we can derive them, because it's a struct and not a newtype, we can't use `RlpEncodableWrapper` etc. If we derive `RlpEncodable`, it will insert an additional header.

Reply:
ah because RlpEncodableWrapper only works with unnamed and can't handle a single field?